File: interestThread.c
#include <pthread.h>
#define NUM_THREADS    10
#define NUM_ITERS      1000000

double monthlyInterest(float balance, float rate, int days)
{
    int i;
    double interest;
    double totalInterest = 0;
    double dailyRate = rate/365;

    for (i = 0; i < days; i++) {
        interest = balance * dailyRate;
        totalInterest += interest;
        balance += interest;
    }

    return totalInterest;
}

void work() {
    int i;
    for (i = 0; i < NUM_ITERS; i++)
        monthlyInterest(50000, 0.03, 30);
}

void *runInterest(void *ignore)
{
    work();
    pthread_exit(NULL);
}

int main(int argc, char *argv[])
{
    pthread_t threads[NUM_THREADS];
    long t;

    for (t = 0; t < NUM_THREADS; t++)
        pthread_create(&threads[t], NULL, work, (void *)NULL);

    pthread_exit(NULL);
    return 0;
}

File: InterestThread.java
public class InterestThread implements Runnable {
    public void run() {
        // 1,000,000,000; i++)
        for (int i = 0; i < 1000000000; i++)
            monthlyInterest(50000, 0.03, 30);
    }

    public static double monthlyInterest(double balance, double apr, int days) {
        double dailyRate = apr/365;
        double totalInterest = 0;
        double interest;

        for (int i = 0; i < days; i++) {
            interest = balance * dailyRate;
            totalInterest += interest;
            balance += interest;
        }

        return totalInterest;
    }

    public static void main(String args[]) {
        long start = System.currentTimeMillis();
        // 1,000,000,000
        for (int i = 0; i < 10000; i++)
            (new Thread(new InterestThread())).start();
        long end = System.currentTimeMillis();
        System.out.println(String.format("%.3f seconds", (float) (end-start)/1000));
    }
}
